gui/src/components/ProviderTokensBalances.tsx

```diff
@@ -0,0 +1,63 @@
+import { useRegisterActions } from "kbar";
+import { ReactNode, createContext } from "react";
+
+import { useAccount } from "../hooks";
+import { useInvoke } from "../hooks/tauri";
+import { useCurrentNetwork } from "../hooks/useCurrentNetwork";
+import { useRefreshTransactions } from "../hooks/useRefreshTransactions";
+import { Address, TokenBalance } from "../types";
+
+interface Value {
+  balances: TokenBalance[];
+  refetchBalances: () => Promise<void>;
+}
+
+export const TokensBalancesContext = createContext<Value>({} as Value);
+
+const actionId = "token-balances";
+
+export function ProviderTokensBalances({ children }: { children: ReactNode }) {
+  const address = useAccount();
+  const { currentNetwork } = useCurrentNetwork();
+  const chainId = currentNetwork?.chain_id;
+
+  const { data: balances, mutate: mutateBalances } = useInvoke<
+    [Address, string][]
+  >("db_get_erc20_balances", { chainId, address });
+
+  const { mutate: refetchTokensBalances } = useInvoke(
+    "alchemy_fetch_balances",
+    { chainId, address },
+    {
+      refreshInterval: 10 * 1000 * 60,
+      revalidateOnFocus: false,
+      revalidateOnMount: false,
+    }
```

Comment:
nice, didn't even consider that SWR could support this already, but looks like a fine way to do it 👍🏼